src/Diagnostics.ModelsAndUtils/Models/Resource/HostingEnvironment.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Diagnostics.ModelsAndUtils.Attributes;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Diagnostics.ModelsAndUtils.Models
+{
+    public class HostingEnvironment : HostingEnvironmentFilter, IResource
+    {
+        /// <summary>
+        /// Name of the Hosting Environment
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Subscription Id(Guid)
+        /// </summary>
+        public string SubscriptionId { get; set; }
+
+        /// <summary>
+        /// Resource Group Name
+        /// </summary>
+        public string ResourceGroup { get; set; }
+
+        /// <summary>
+        /// Internal Name (For Example:- waws-prod-....)
+        /// </summary>
+        public string InternalName { get; set; }
+
+        /// <summary>
+        /// Service Address
+        /// </summary>
```

Comment:
Add VNET name for ASE ?

Reply:
I can worry about adding additional things for ASE, this filter work is a big work item alone. Lets just get this out there.

Reply:
ya :)

Reply:
Well we have the vnet name at least...other detectors can get this and get vnet info through the geoMaster provider I guess...